effekt/jvm/src/test/scala/effekt/ParserTests.scala

```diff
@@ -928,6 +928,58 @@ class ParserTests extends munit.FunSuite {
         |""".stripMargin)
   }
 
+  test("Namespaces") {
+    parseDefinitions(
+      """val x = 4
+        |val y = 5
+        |""".stripMargin)
+
+    val nested = parseToplevel(
+      """namespace list {
+        |  val x = 4
+        |  val y = 5
+        |}
+        |""".stripMargin)
+
+    val semi = parseToplevel(
+      """namespace list;
+        |val x = 4
+        |val y = 5
+        |""".stripMargin)
+
+    assertEqualModuloSpans(nested, semi)
+
+    val nested2 = parseProgram(
+      """namespace list {
+        |  namespace internal {
+        |
+        |    val x = 4
+        |    val y = 5
+        |  }
+        |}
+        |""".stripMargin)
+
+    val semi2 = parseProgram(
+      """namespace list;
+        |namespace internal;
+        |
+        |val x = 4
+        |val y = 5
+        |""".stripMargin)
+
+    val semiInsertion = parseProgram(
+      """namespace list
+        |namespace internal
+        |
+        |val x = 4
+        |val y = 5
+        |""".stripMargin)
+
+    // TODO: how to do equality on Programs?
```

Comment:
How can we do equality on Programs (`ModuleDecl`)? Previously we did parseDefinition and `assertEqualModuloSpans`. `(a zip b).forEach(assertEqualModuloSpans)` does not work. Should I remove the equality assertions?